post/views.py

```diff
@@ -26,10 +28,26 @@ def post(self, request, *args, **kwargs):
 
 
 class PostDetailView(View):
+    form_class = CommentCreateForm
+
+    def setup(self, request, *args, **kwargs):
+        self.post_instance = get_object_or_404(Post, pk=kwargs['post_id'], slug=kwargs['post_slug'])
+        return super().setup(self, request, *args, **kwargs)
+
     def get(self, request, post_id, post_slug):
-        post = get_object_or_404(Post, pk=post_id, slug=post_slug)
-        comments = post.pcomments.filter(is_reply=False)
-        return render(request, 'post/detail.html', {'post': post, 'comments': comments})
+        comments = self.post_instance.pcomments.filter(is_reply=False)
+        return render(request, 'post/detail.html', {'post': self.post_instance, 'comments': comments, 'form': self.form_class})
+
+    @method_decorator(login_required)
```

Comment:
**method_decorator** for method post beacuase 
only login users can post comment